flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskLocalStateStoreImpl.java

```diff
@@ -300,6 +295,34 @@ private void deleteDirectory(File directory) throws IOException {
 		}
 	}
 
+	/**
+	 * Pruning the useless checkpoints, it should be called only when holding the {@link #lock}.
+	 */
+	private void pruneCheckpoints(Predicate<Long> pruningChecker, boolean breakOnceCheckerFalse) {
```

Comment:
We can use `LongPredicate` instead of `Predicate<Long>`

Reply:
👍 I prefer to use `LongPredicate`, addressing ...